examples/shouldi/shouldi/safety.py

```diff
@@ -29,16 +29,15 @@
         "check",
         "--stdin",
         "--json",
-        stdin=asyncio.subprocess.PIPE,
-        stdout=asyncio.subprocess.PIPE,
-        stderr=asyncio.subprocess.PIPE,
+        stdin=asyncio.subprocess.Process.PIPE,
+        stdout=asyncio.subprocess.Process.PIPE,
+        stderr=asyncio.subprocess.Process.PIPE,
     )
 
-    proc.stdin.write(pinned.encode())
-    proc.stdin.write(b"\n")
-    proc.stdin.close()
-
-    stdout, _stderr = await proc.communicate()
+    
+    stdout, _stderr = await proc.communicate([pinned.encode(),b"\n"])
```

Comment:
This should just be `proc.communicate(pinned.encode() + b"\n")`. Also, we try to avoid unnecessary additions of blank lines (so please the one empty line added before and the two empty lines added after).

Not sure if you've got the development environment setup, but if not then you might like gitpod, https://gitpod.io/#https://github.com/intel/dffml once it finishes installing you can run:

```console
$ git remote add yash https://github.com/Yash-Varshney/dffml.git
$ git fetch yash
remote: Enumerating objects: 27, done.
remote: Counting objects: 100% (27/27), done.
remote: Compressing objects: 100% (19/19), done.
remote: Total 32 (delta 16), reused 11 (delta 8), pack-reused 5
Unpacking objects: 100% (32/32), done.
From https://github.com/Yash-Varshney/dffml
 * [new branch]      Yash-Varshney-patch-1 -> yash/Yash-Varshney-patch-1
 * [new branch]      Yash-Varshney-patch-2 -> yash/Yash-Varshney-patch-2
 * [new branch]      binsec                -> yash/binsec
 * [new branch]      encoding              -> yash/encoding
 * [new branch]      gh-pages              -> yash/gh-pages
 * [new branch]      master                -> yash/master
$ git branch --set-upstream-to yash/master
Branch 'master' set up to track remote branch 'master' from 'yash'.
$ git pull
Merge made by the 'recursive' strategy.
 examples/shouldi/shouldi/safety.py | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)
$ git status
On branch master
Your branch is ahead of 'yash/master' by 34 commits.
  (use "git push" to publish your local commits)

nothing to commit, working tree clean
$ (cd examples/shouldi/ && python setup.py test -s tests.test_safety)
running test
running egg_info
writing shouldi.egg-info/PKG-INFO
writing dependency_links to shouldi.egg-info/dependency_links.txt
writing entry points to shouldi.egg-info/entry_points.txt
writing requirements to shouldi.egg-info/requires.txt
writing top-level names to shouldi.egg-info/top_level.txt
reading manifest file 'shouldi.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'shouldi.egg-info/SOURCES.txt'
running build_ext
test_run (tests.test_safety.TestSafetyCheck) ... ERROR

======================================================================
ERROR: test_run (tests.test_safety.TestSafetyCheck)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspace/dffml/dffml/util/asynctestcase.py", line 51, in run_it
    result = self.loop.run_until_complete(coro(*args, **kwargs))
  File "/home/gitpod/.pyenv/versions/3.7.6/lib/python3.7/asyncio/base_events.py", line 583, in run_until_complete
    return future.result()
  File "/workspace/dffml/examples/shouldi/tests/test_safety.py", line 8, in test_run
    results = await safety_check("insecure-package", "0.1.0")
  File "/workspace/dffml/examples/shouldi/shouldi/safety.py", line 32, in safety_check
    stdin=asyncio.subprocess.Process.PIPE,
AttributeError: type object 'Process' has no attribute 'PIPE'

----------------------------------------------------------------------
Ran 1 test in 0.004s

FAILED (errors=1)
Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
error: Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
```

Then you'll just need to commit the changes with `git commit` and push them with `git push`